cpu/cc2538/periph/gpio.c

```diff
@@ -213,10 +213,11 @@ static inline void handle_isr(uint8_t port_num)
     port->IC             = 0x000000ff;
     port->IRQ_DETECT_ACK = (0xff << (port_num * GPIO_BITS_PER_PORT));
 
-    for (int i = 0; i < GPIO_BITS_PER_PORT; i++) {
-        if (state & (1 << i)) {
-            isr_ctx[port_num][i].cb(isr_ctx[port_num][i].arg);
-        }
+    state &= GPIO_BIT_MASK;
+    while (state) {
```

Comment:
if using builtins, wouldn't sth like:

```
int pin;
while ((pin = __builtin_ffs(state))) {
  isr_ctx[port_num][pin].cb(isr_ctx[port_num][pin].arg);
  state &= ~(1 << pin);
}
```

... save some arithetic?

Reply:
The Cortex-M3 only has `clz`, no `ffs`
http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.100165_0201_00_en/ric1414056333562.html

Reply:
but gcc's __builtin_ffs() is implemented using clz?

Reply:
Want to play a [game of golf](https://godbolt.org/z/Lfo23M)? 😉

Reply:
Godbolt is awesome! Thanks! :wink: